mesonbuild/compilers/compilers.py

```diff
@@ -113,6 +113,13 @@ def is_library(fname):
                           'release': ['-O3'],
                           'minsize': ['-Os', '-g']}
 
+arm_buildtype_args = {'plain': [],
+                      'debug': ['-O0', '--debug'],
+                      'debugoptimized': ['-O1', '--debug'],
```

Comment:
Why is this -O1 instead of -O2? Is ARM somehow nonstandard?

Reply:
In our experience with ARM compiler, -O1 option does restricted optimizations for debugging and O2 will provide high optimization which is not suitable to debug with a debugger. Hence the -O1 level for debug Optimized.